src/streams/EditStreamCard.js

```diff
@@ -26,6 +31,151 @@ type Props = $ReadOnly<{|
   onComplete: (name: string, description: string, privacy: Privacy) => void | Promise<void>,
 |}>;
 
+function useStreamPrivacyOptions(initialValue: Privacy) {
+  const {
+    webPublicStreamsEnabled,
+    enableSpectatorAccess,
+    createWebPublicStreamPolicy,
+    name: realmName,
+  } = useSelector(getRealm);
+  const canCreateWebPublicStreams = useSelector(getCanCreateWebPublicStreams);
+  const { role } = useSelector(getOwnUser);
+  const realmUrl = useSelector(getRealmUrl);
+
+  return useMemo(
+    () =>
+      [
+        webPublicStreamsEnabled && enableSpectatorAccess
+          ? {
+              key: 'web-public',
+              title: 'Web-public',
```

Comment:
nit: Can make the contents here a couple of levels less deeply indented like so:
```suggestion
  const webPublicStreamsActuallyEnabled = webPublicStreamsEnabled && enableSpectatorAccess;
  return useMemo(
    () =>
      [
        webPublicStreamsActuallyEnabled && {
          key: 'web-public',
          title: 'Web-public',
```

Reply:
Hmm, I guess this one runs into Flow trouble because the type on `Array#filter(Boolean)` isn't specific enough to say that it filters out `false`. From flowlib/core.js (gotten by jump-to-definition on the `filter` call):
```js
    filter(callbackfn: typeof Boolean): Array<$NonMaybeType<T>>;
```

I think I still like putting the one-line "nothing here" case first, though, akin to an early return. Like this:
```suggestion
  return useMemo(
    () =>
      [
        !(webPublicStreamsEnabled && enableSpectatorAccess)
          ? undefined
          : {
              key: 'web-public',
              title: 'Web-public',
```

Reply:
> Hmm, I guess this one runs into Flow trouble because the type on `Array#filter(Boolean)` isn't specific enough to say that it filters out `false`.

Oh, hmm, yeah.